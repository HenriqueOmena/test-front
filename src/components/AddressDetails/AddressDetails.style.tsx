import { makeStyles } from "@material-ui/core";

const AddressDetailsStyle = makeStyles(() => ({
  address: {
    width: 250,
    "& p": {
      display: "flex",
      alignItems: "center"
    },
    "& span": {
      marginLeft: 10
    }
  }
}));

export default AddressDetailsStyle;
