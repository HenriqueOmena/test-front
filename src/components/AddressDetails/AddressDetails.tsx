import React from "react";
import LocationCityIcon from "@material-ui/icons/LocationCity";
import LocationOnIcon from "@material-ui/icons/LocationOn";
import HomeIcon from "@material-ui/icons/Home";
import MapIcon from "@material-ui/icons/Map";
import { IAddress } from "components/DoctorCard/DoctorCard.interface";
import AddressDetailsStyle from "./AddressDetails.style";

const AddressDetails = ({ city, country, line1, line2, postalCode }: IAddress) => {
  const classes = AddressDetailsStyle();

  return (
    <div className={classes.address}>
      <p>
        <MapIcon />
        <span>{city + ", " + country}</span>
      </p>
      <p>
        <HomeIcon />
        <span>{line1}</span>
      </p>
      <p>
        <LocationCityIcon />
        <span>{line2}</span>
      </p>
      <p>
        <LocationOnIcon />
        <span>{postalCode}</span>
      </p>
    </div>
  );
};

export default AddressDetails;
