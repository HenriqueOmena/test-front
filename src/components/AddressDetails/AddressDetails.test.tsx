import { render } from "@testing-library/react";
import AddressDetails from "./AddressDetails";

describe("Component - AddressDetails", () => {
  const propMock = {
    city: "Funchal",
    country: "Portugal",
    line1: "Santa",
    line2: "Rita",
    postalCode: "0099"
  };

  let component = <AddressDetails {...propMock} />;

  it("Should Match SnapShot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should render each prop", async () => {
    const { getByText } = render(component);
    const { city, country, line1, line2, postalCode } = propMock;
    const cityCountryText = getByText(city + ", " + country);
    const line1Text = getByText(line1);
    const line2Text = getByText(line2);
    const postalCodeText = getByText(postalCode);

    expect(cityCountryText).toBeVisible();
    expect(line1Text).toBeVisible();
    expect(line2Text).toBeVisible();
    expect(postalCodeText).toBeVisible();
  });
});
