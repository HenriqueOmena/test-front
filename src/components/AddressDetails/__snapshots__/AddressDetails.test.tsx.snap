// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AddressDetails Should Match SnapShot 1`] = `
<AddressDetails
  city="Funchal"
  country="Portugal"
  line1="Santa"
  line2="Rita"
  postalCode="0099"
/>
`;
